### DISCUSSION TITLE
Missing libraries during Sonar Analysis

### DISCUSSION THREAD

**Discussion by phermsdorf** [eclipse-tycho/tycho#1086] [General]
Hi,

during SonarQube Analysis (after the tycho build) i see warnings like these:

> 00:43:03.027  [INFO] Did not optimize analysis for any files, performed a full analysis for all 10 files.
> 00:43:03.027  [WARNING] Dependencies/libraries were not provided for analysis of SOURCE files. The 'sonar.java.libraries' property is empty. Verify your configuration, as you might end up with less precise results.
> 

and 

> 00:57:07.140  Status ERROR: org.eclipse.jdt.core code=4 Could not retrieve declared fields org.eclipse.jdt.internal.compiler.problem.AbortCompilation: Pb(324) The type org.eclipse.swt.widgets.Label cannot be resolved. It is indirectly referenced from required .class files

SonarAnalysis is basically a `mvn sonar:sonar` run.

I wanted to pass a directory to `sonar.java.libraries` where all bundles from the resolved target platform are, but i don't know where tycho does resolve the target to.
(like the PDE `ws/.metadata/.plugins/org.eclipse.pde.core/.bundle_pool/plugins/` folder)

Any hint or ideas are very welcome.

Thanks!

Bye Peter

**Comment by Bananeweizen** [comment]
I've never used that property in any of the (company) Sonar scans that I run. I didn't really miss anything until now, but of course there might be a small amount of rules that would require the additional information. And given that it already takes several _hours_ to analyze 500 KLOC for us, I don't really want an even deeper analysis...
